be/src/runtime/routine_load/data_consumer_group.cpp

```diff
@@ -170,22 +170,28 @@ Status KafkaDataConsumerGroup::start_all(StreamLoadContext* ctx) {
             VLOG(3) << "get kafka message"
                     << ", partition: " << msg->partition() << ", offset: " << msg->offset() << ", len: " << msg->len();
 
-            Status st = (kafka_pipe.get()->*append_data)(static_cast<const char*>(msg->payload()),
-                                                         static_cast<size_t>(msg->len()), row_delimiter);
-
-            if (st.ok()) {
-                received_rows++;
-                left_bytes -= msg->len();
-                cmt_offset[msg->partition()] = msg->offset();
-                VLOG(3) << "consume partition[" << msg->partition() << " - " << msg->offset() << "]";
+            if (msg->err() == RdKafka::ERR__PARTITION_EOF) {
+                if (msg->offset() > 0) {
+                    cmt_offset[msg->partition()] = msg->offset() - 1;
```

Comment:
What would happen if the last offset is empty or non-empty ?

Reply:
No effect

Reply:
FE will only update the Partition corresponding to the Offset I returned